kytos/core/user.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+from optparse import Option
+from typing import Literal, Optional
+from pydantic import BaseModel, EmailStr, Field, KafkaDsn, constr, validator
+
+
+class DocumentBaseModel(BaseModel):
+    """DocumentBaseModel"""
+
+    id: str = Field(None, alias="_id")
+    inserted_at: Optional[datetime]
+    updated_at: Optional[datetime]
+    deleted_at: Optional[datetime]
+
+    def dict(self, **kwargs) -> dict:
+        """Model to dict."""
+        values = super().dict(**kwargs)
+        if "id" in values and values["id"]:
+            values["_id"] = values["id"]
+        if "exclude" in kwargs and "_id" in kwargs["exclude"]:
+            values.pop("_id")
+        return values
+
+
+class UserDoc(DocumentBaseModel):
+    """UserDocumentModel."""
+
+    full_name: str
+    state: Literal['active', 'inactive'] = 'active'
+    password: constr(min_length=8)
+    email: EmailStr
+
+    @validator('password', pre=True, always=True)
+    def have_digit_letter(cls, password):
+        """Check if password has at least a letter and a number"""
```

Comment:
@Alopalao if you could double check the business requirements here https://github.com/kytos-ng/kytos/issues/294, we'd like to have an upper char in the password, make sure to cover this with unit test later on as you make more progress on this PR.

Reply:
Yeah, my bad. Updated.

Reply:
The new commits looks great, it's nice to have this descriptive error message too. The name of the method is still using the old one, this method is doing more than just `have_digit_letter` maybe we can name it more generically like `validate_password` or something else similar that you might want to use.

Reply:
Changed